iac-extensions/arm/src/main/java/org/sonar/iac/arm/visitors/ArmHighlightingVisitor.java

```diff
@@ -0,0 +1,121 @@
+      highlight(tree.getExisting(), KEYWORD);
+    });
+    register(DecoratorImpl.class, (ctx, tree) -> {
+      highlight(tree.keyword(), ANNOTATION);
+      Expression expression = tree.expression();
+      if (expression instanceof FunctionCall) {
+        highlight(((FunctionCall) expression).name(), ANNOTATION);
+      }
+    });
+    register(TargetScopeDeclarationImpl.class, (ctx, tree) -> highlight(tree.keyword(), KEYWORD));
+    register(ImportDeclarationImpl.class, (ctx, tree) -> highlight(tree.keyword(), KEYWORD));
+    register(ImportWithClause.class, (ctx, tree) -> highlight(tree.keyword(), KEYWORD));
+    register(ImportAsClause.class, (ctx, tree) -> highlight(tree.keyword(), KEYWORD));
+    register(ParameterDeclarationImpl.class, (ctx, tree) -> {
+      highlight(tree.keyword(), KEYWORD);
+      highlight(tree.identifier(), KEYWORD_LIGHT);
+    });
+    register(TypeDeclarationImpl.class, (ctx, tree) -> {
+      highlight(tree.keyword(), KEYWORD);
+      highlight(tree.name(), KEYWORD_LIGHT);
+    });
+    register(AbstractDeclaration.class, (ctx, tree) -> {
+      highlight(tree.keyword(), KEYWORD);
+      highlight(tree.identifier(), KEYWORD_LIGHT);
+    });
+    register(ModuleDeclarationImpl.class, (ctx, tree) -> {
+      highlight(tree.keyword(), KEYWORD);
+      highlight(tree.name(), KEYWORD_LIGHT);
+    });
+    register(OutputDeclarationImpl.class, (ctx, tree) -> {
+      highlight(tree.keyword(), KEYWORD);
+      highlight(tree.name(), KEYWORD_LIGHT);
+      highlight(tree.type(), KEYWORD);
+    });
+    register(FunctionDeclarationImpl.class, (ctx, tree) -> {
+      highlight(tree.keyword(), KEYWORD);
+      highlight(tree.name(), KEYWORD_LIGHT);
+    });
+    register(IfConditionImpl.class, (ctx, tree) -> highlight(tree.keyword(), KEYWORD));
+    register(ForExpressionImpl.class, (ctx, tree) -> {
+      highlight(tree.forKeyword(), KEYWORD);
+      highlight(tree.inKeyword(), KEYWORD);
+    });
+    register(ForVariableBlockImpl.class, (ctx, tree) -> highlight(tree.itemIdentifier(), ANNOTATION));
+    register(LocalVariableImpl.class, (ctx, tree) -> highlight(tree.identifier(), ANNOTATION));
+    register(PropertyImpl.class, (ctx, tree) -> highlight(tree.key(), ANNOTATION));
+
+    register(AmbientTypeReference.class, (ctx, tree) -> highlight(tree, KEYWORD));
+
+    register(BooleanLiteralImpl.class, (ctx, tree) -> highlight(tree.children().get(0), CONSTANT));
+    register(NullLiteralImpl.class, (ctx, tree) -> highlight(tree.children().get(0), CONSTANT));
+    register(NumericLiteralImpl.class, (ctx, tree) -> highlight(tree.children().get(0), CONSTANT));
+
+    register(InterpolatedStringImpl.class, (ctx, tree) -> highlight(tree, STRING));
+    register(MultilineStringImpl.class, (ctx, tree) -> highlight(tree, STRING));
+    register(StringCompleteImpl.class, (ctx, tree) -> highlight(tree, STRING));
+    register(StringLiteralImpl.class, (ctx, tree) -> highlight(tree, STRING));
```

Comment:
What about a `register` method, which accepts multiple classes?

Reply:
This can be impossible due to Java generics...